src/Email/EmailServiceProvider.php

```diff
@@ -31,6 +36,31 @@ public function register(Application $app)
                 $config['email']['mandrill_api_key'] = getenv('MANDRILL_API_KEY');
             }
         }
+        if (array_key_exists('mailgun_api_key', $config['email'])) {
+            $app['mailgun'] = new Mailgun($config['email']['mailgun_api_key']);
+            if (! Arr::path($config, 'email.template_path')) {
+                Arr::setPath(
+                    $config,
+                    'email.template_path',
+                    APP_PATH . '/templates'
+                );
+            }
+            $app['handlebars'] = new Handlebars([
```

Comment:
As this is specific to the email service, change it to `email-handlebars` or something like that and as said before, change it to $app->share()